accounts/src/store/accounts_db.rs

```diff
@@ -0,0 +1,829 @@
+use solana_rpc_client_api::filter::RpcFilterType;
+use solana_sdk::account::{Account, AccountSharedData, ReadableAccount};
+use solana_sdk::clock::Slot;
+use solana_sdk::genesis_config::ClusterType;
+use solana_sdk::pubkey::Pubkey;
+use solana_sdk::transaction_context::TransactionAccount;
+
+use solana_lite_rpc_core::commitment_utils::Commitment;
+use solana_lite_rpc_core::commitment_utils::Commitment::Processed;
+use solana_lite_rpc_core::structures::account_data::AccountData;
+use Commitment::{Confirmed, Finalized};
+
+use crate::account_store_interface::{AccountLoadingError, AccountStorageInterface};
+
+// FIXME what are all those configs
+pub const BINS: usize = 8192;
+pub const FLUSH_THREADS: usize = 1;
+
+pub const ACCOUNTS_INDEX_CONFIG: AccountsIndexConfig = AccountsIndexConfig {
+    bins: Some(BINS),
+    flush_threads: Some(FLUSH_THREADS),
+    drives: None,
+    index_limit_mb: IndexLimitMb::Unspecified,
+    ages_to_stay_in_cache: None,
+    scan_results_limit_bytes: None,
+    started_from_validator: false,
+};
+
+pub const ACCOUNTS_DB_CONFIG: AccountsDbConfig = AccountsDbConfig {
+    index: Some(ACCOUNTS_INDEX_CONFIG),
+    base_working_path: None,
+    accounts_hash_cache_path: None,
+    shrink_paths: None,
+    read_cache_limit_bytes: None,
+    write_cache_limit_bytes: None,
+    ancient_append_vec_offset: None,
+    skip_initial_hash_calc: false,
+    exhaustively_verify_refcounts: false,
+    create_ancient_storage: CreateAncientStorage::Pack,
+    test_partitioned_epoch_rewards: TestPartitionedEpochRewards::None,
+    test_skip_rewrites_but_include_in_bank_hash: false,
+    storage_access: StorageAccess::Mmap,
+};
+
+pub struct AccountsDb {
+    accounts: Accounts,
+    processed_slot: AtomicU64,
+    confirmed_slot: AtomicU64,
+    finalised_slot: AtomicU64,
+}
+
+impl AccountsDb {
+    pub fn new() -> Self {
+        let db = SolanaAccountsDb::new_with_config(
+            vec![],
+            &ClusterType::MainnetBeta,
+            AccountSecondaryIndexes::default(),
+            AccountShrinkThreshold::default(),
+            Some(ACCOUNTS_DB_CONFIG),
+            None,
+            Arc::default(),
+        );
+
+        let accounts = Accounts::new(Arc::new(db));
+        Self {
+            accounts,
+            processed_slot: AtomicU64::new(0),
+            confirmed_slot: AtomicU64::new(0),
+            finalised_slot: AtomicU64::new(0),
+        }
+    }
+
+    pub fn new_for_testing() -> Self {
+        let db = SolanaAccountsDb::new_single_for_tests();
+        let accounts = Accounts::new(Arc::new(db));
+        Self {
+            accounts,
+            processed_slot: AtomicU64::new(0),
+            confirmed_slot: AtomicU64::new(0),
+            finalised_slot: AtomicU64::new(0),
+        }
+    }
+}
+
+#[async_trait]
+impl AccountStorageInterface for AccountsDb {
+    async fn update_account(&self, account_data: AccountData, _commitment: Commitment) -> bool {
```

Comment:
note: we actually do not want to have and support commitments other than PROCESSED as this will push complexity of buffering and slot commitment handling to yellowstone-grpc